ci-operator/step-registry/cucushift/installer/check/azure/publish-mixed/cucushift-installer-check-azure-publish-mixed-commands.sh

```diff
@@ -0,0 +1,189 @@
+        cat /usr/lib64/az/lib/python*/site-packages/certifi/cacert.pem >> /tmp/ca.pem
+        export REQUESTS_CA_BUNDLE=/tmp/ca.pem
+    fi
+    az cloud register \
+        -n ${cloud_name} \
+        --endpoint-resource-manager "${AZURESTACK_ENDPOINT}" \
+        --suffix-storage-endpoint "${SUFFIX_ENDPOINT}"
+    az cloud set --name ${cloud_name}
+    az cloud update --profile 2019-03-01-hybrid
+else
+    az cloud set --name AzureCloud
+fi
+az login --service-principal -u "${AZURE_AUTH_CLIENT_ID}" -p "${AZURE_AUTH_CLIENT_SECRET}" --tenant "${AZURE_AUTH_TENANT_ID}" --output none
+
+INSTALL_CONFIG="${SHARED_DIR}/install-config.yaml"
+API_PUBLISH_STRATEGY=$(yq-go r "${INSTALL_CONFIG}" 'operatorPublishingStrategy.apiserver')
+INGRESS_PUBLISH_STRATEGY=$(yq-go r "${INSTALL_CONFIG}" 'operatorPublishingStrategy.ingress')
+INFRA_ID=$(jq -r .infraID ${SHARED_DIR}/metadata.json)
+BASE_DOMAIN=$(yq-go r "${INSTALL_CONFIG}" 'baseDomain')
+BASE_DOMAIN_RESOURCE_GROUP=$(yq-go r "${INSTALL_CONFIG}" 'platform.azure.baseDomainResourceGroupName')
+CLUSTER_NAME=$(yq-go r "${INSTALL_CONFIG}" 'metadata.name')
+RESOURCE_GROUP=$(yq-go r "${INSTALL_CONFIG}" 'platform.azure.resourceGroupName')
+if [[ -z "${RESOURCE_GROUP}" ]]; then
+    RESOURCE_GROUP="${INFRA_ID}-rg"
+fi
+
+PUBLIC_LB_NAME="${INFRA_ID}"
+INTERNAL_LB_NAME="${INFRA_ID}-internal"
+API_SERVER_PORT="6443"
+INGRESS_PORTS="80 443"
+API_DNS_NAME="api.${CLUSTER_NAME}"
+INGRESS_DNS_NAME="*.apps.${CLUSTER_NAME}"
+
+check_result=0
+#save lb info
+echo "lb rules on public loadbalancer: "
+public_lb_rules=$(mktemp)
+az network lb rule list --lb-name ${PUBLIC_LB_NAME} -g ${RESOURCE_GROUP} | tee ${public_lb_rules}
+
+echo -e "\nlb rules on internal loadbalancer: "
+internal_lb_rules=$(mktemp)
+az network lb rule list --lb-name ${INTERNAL_LB_NAME} -g ${RESOURCE_GROUP} | tee ${internal_lb_rules}
+
+if [[ "${API_PUBLISH_STRATEGY}" == "Internal" ]]; then
+    # Load Balance Rules on port 6443 should only be created in internal LB
+    # outbound rule is created in public LB
+    # API dns record is not created in public zone 
+
+    echo "API_PUBLISH_STRATEGY: ${API_PUBLISH_STRATEGY}"
+    echo "(1/4) check that LB rule on port ${API_SERVER_PORT} is not created in public LB..."
+    api_pubilc_lb_rule=$(cat ${public_lb_rules} | jq -r ".[] | select(.backendPort==${API_SERVER_PORT})")
+    if [[ ! -z "${api_pubilc_lb_rule}" ]]; then
+        echo "ERROR: found lb rule on port ${API_SERVER_PORT} in public LB!"
+        check_result=1
+    fi
+
+    echo "(2/4) check that LB rule on port ${API_SERVER_PORT} is created in internal LB..."
+    api_internal_lb_rule=$(cat ${internal_lb_rules} | jq -r ".[] | select(.backendPort==${API_SERVER_PORT})")
+    if [[ -z "${api_internal_lb_rule}" ]]; then
+        echo "ERROR: not found lb rule on port ${API_SERVER_PORT} in internal LB!"
+        check_result=1
+    fi
+
+    echo "(3/4) check that outbound rule is created in public zone..."
```

Comment:
Do we need to check public ip address is configured in public lb?

Reply:
good point, updated.